src/Tribe/Image/Uploader.php

```diff
@@ -77,105 +113,138 @@ protected function upload_file( $file_url ) {
 		 *
+		 * @since 4.9.5
+		 *
 		 * @param bool   $allow_local_urls Whether to allow local URLs.
 		 * @param string $file_url         File URL.
-		 *
-		 * @since 4.9.5
 		 */
 		$allow_local_urls = apply_filters( 'tribe_image_uploader_local_urls', false, $file_url );
 
-		if ( ! filter_var( $file_url, FILTER_VALIDATE_URL ) && ! $allow_local_urls ) {
+		if ( ! $allow_local_urls && ! filter_var( $file_url, FILTER_VALIDATE_URL ) ) {
 			return false;
 		}
 
-		/*
-		 * Since `file_get_contents` would fail silently we set an explicit
-		 * error handler to catch the content of error.s.
-		 */
-		set_error_handler( array( $this, 'handle_error' ) );
+		$is_local = false;
+		// This is a local file no need to fetch it from the wire.
+		if ( $allow_local_urls && file_exists( $file_url ) ) {
+			$file = $file_url;
+			$is_local = true;
+		} else {
+			/**
+			 * Some CDN services will append query arguments to the image URL; removing
+			 * them now has the potential of blocking the image fetching completely so we
+			 * let them be here.
+			 */
+			$file = download_url( $file_url );
+			if ( is_wp_error( $file ) ) {
+				do_action( 'tribe_log', 'error', __CLASS__, [
+					'message' => $file->get_error_message(),
+					'url'     => $file_url,
+					'error'   => $file,
+				] );
+
+				return false;
+			}
+		}
 
-		/*
-		 * Some CDN services will append query arguments to the image URL; removing
-		 * them now has the potential of blocking the image fetching completely so we
-		 * let them be here.
-		 */
-		try {
-			$contents = file_get_contents( $file_url );
-		} catch ( Exception $e ) {
-			$message = sprintf( 'Could not upload image file "%s": with message "%s"', $file_url, $e->getMessage() );
-			tribe( 'logger' )->log_error( $message, 'Image Uploader' );
+		// Upload file into WP and leave WP handle the resize and such.
+		$attachment_id = media_handle_sideload( [
+			'name'           => $this->create_file_name( $file ),
+			'tmp_name'       => $file,
+			'post_mime_type' => 'image',
+		] );
+
+		// Remove the temporary file as is no longer required at this point.
+		if ( ! $is_local && file_exists( $file ) ) {
+			@unlink( $file );
+		}
 
-			restore_error_handler();
+		if ( is_wp_error( $attachment_id ) ) {
+			do_action( 'tribe_log', 'error', __CLASS__, [
+				'message' => $attachment_id->get_error_message(),
+				'url'     => $file_url,
+				'error'   => $attachment_id,
+			] );
 
 			return false;
 		}
 
-		restore_error_handler();
+		update_post_meta( $attachment_id, '_tribe_importer_original_url', $file_url );
 
-		if ( false === $contents ) {
-			$message = sprintf( 'Could not upload image file "%s": failed getting the contents.', $file_url );
-			tribe( 'logger' )->log_error( $message, 'Image Uploader' );
+		$this->maybe_init_attachment_guids_cache();
+		$this->maybe_init_attachment_original_urls_cache();
 
-			return false;
-		}
+		self::$attachment_guids_cache[ get_post( $attachment_id )->guid ] = $attachment_id;
```

Comment:
We should save this into a variable and check before using it.

Reply:
We have a check for this a little bit above.

Here: https://github.com/moderntribe/tribe-common/blob/eeac34c30b926568356eeff5c18c5fe8f3f43c7d/src/Tribe/Image/Uploader.php#L162